### PR TITLE
Setuid: Add functional tests for prioritization

### PR DESCRIPTION
**Pull request by marki1an**
Test for  https://github.com/prebid/prebid-server-java/pull/2085 and https://github.com/prebid/prebid-server-java/pull/2089

### CODE REVIEW

File: src/test/groovy/org/prebid/server/functional/tests/SetUidSpec.groovy [hunk 1]
```diff
@@ -31,4 +54,177 @@ class SetUidSpec extends BaseSpec {
...
+        }
+
+        when: "PBS processes setuid request"
+        def response = prebidServerService.sendSetUidRequest(request, uidsCookie)
+
+        then: "Response shouldn't contain uids cookie"
+        assert response.uidsCookie.tempUIDs.size() == 0
+    }
+
+    def "PBS setuid should populate uids cookie with non priority bidder"() {
```
Comment (Net-burst) [inline src/test/groovy/org/prebid/server/functional/tests/SetUidSpec.groovy:75]: This test checks for a different thing. You're not populating anything.

File: src/test/groovy/org/prebid/server/functional/tests/SetUidSpec.groovy [hunk 2]
```diff
@@ -31,4 +54,177 @@ class SetUidSpec extends BaseSpec {
...
+        def uidsCookie = UidsCookie.defaultUidsCookie.tap {
+            tempUIDs = [(APPNEXUS): defaultUidWithExpiry,
+                        (RUBICON) : defaultUidWithExpiry]
+        }
+
+        when: "PBS processes setuid request"
+        def response = prebidServerService.sendSetUidRequest(request, uidsCookie)
+
+        then: "Response should contain uids cookies"
+        assert response.uidsCookie.tempUIDs[GENERIC]
```
Comment (Net-burst) [inline src/test/groovy/org/prebid/server/functional/tests/SetUidSpec.groovy:112]: And what about other uids? Were they removed? What happened to them?

File: src/test/groovy/org/prebid/server/functional/tests/SetUidSpec.groovy [hunk 3]
```diff
@@ -31,4 +54,177 @@ class SetUidSpec extends BaseSpec {
         assert response.responseBody ==
                 ResourceUtil.readByteArrayFromClassPath("org/prebid/server/functional/tracking-pixel.png")
     }
+
+    def "PBS setuid should remove expired uids cookie"() {
+        given: "Default SetuidRequest"
+        def request = SetuidRequest.defaultSetuidRequest
```
Comment (Net-burst) [inline src/test/groovy/org/prebid/server/functional/tests/SetUidSpec.groovy:60]: Hm. Our default request doesn't contain a uuid, which is kinda mandatory. Looks like this should be changed.

File: src/test/groovy/org/prebid/server/functional/tests/SetUidSpec.groovy [hunk 4]
```diff
@@ -31,4 +54,177 @@ class SetUidSpec extends BaseSpec {
...
+        PrebidServerService prebidServerService = pbsServiceFactory.getService(PBS_CONFIG +
+                ["cookie-sync.pri"                  : GENERIC.value])
+
+        and: "Setuid request"
+        def request = SetuidRequest.defaultSetuidRequest.tap {
+            uid = UUID.randomUUID().toString()
+        }
+        def uidsCookie = UidsCookie.defaultUidsCookie.tap {
+            def uidWithExpiry = defaultUidWithExpiry.tap {
+                expires = ZonedDateTime.now(Clock.systemUTC()).plusDays(2)
```
Comment (Net-burst) [inline src/test/groovy/org/prebid/server/functional/tests/SetUidSpec.groovy:126]: This should be oldest uids, right? Please differentiate both uids somehow. You're relying on clock to be really granular.

**Review by Net-burst — CHANGES_REQUESTED** [review]

File: src/test/groovy/org/prebid/server/functional/tests/SetUidSpec.groovy (hunk 3 above)
Comment (marki1an) [reply src/test/groovy/org/prebid/server/functional/tests/SetUidSpec.groovy:62]: I don't think so,if it's was mandatory then we could get a error when we try request with empty uid

Comment (Net-burst) [reply src/test/groovy/org/prebid/server/functional/tests/SetUidSpec.groovy:60]: There won't be an error, but PBS also won't return any UIDs too.

**Review by Net-burst — APPROVED** [review]
LGTM

**Pull request merged**